### PR TITLE
[Aggregate] Correctly handle ordering multiple fields; don't crash on nil

### LINKED ISSUE #144: [Aggregate] `ORDER BY` gives priority to proceeding clauses
Related to #140. `ARRAY_AGG`, `ARRAY_CONCAT_AGG` and `STRING_AGG` functions do not correctly implement ordering

### PR DESCRIPTION
**Pull request by ohaibbq**
Closes goccy/go-zetasqlite#145
Closes goccy/bigquery-emulator#198
Closes #144

### CODE REVIEW

File: query_test.go [hunk 1]
```diff
@@ -668,6 +675,13 @@ SELECT ARRAY_CONCAT_AGG(x) AS array_concat_agg FROM (
 				[]interface{}{nil, int64(1), int64(2), int64(3), int64(4), int64(5), int64(6), int64(7), int64(8), int64(9)},
 			}},
 		},
+		{
+			name:  "array_concat_agg with null in order by",
+			query: `WITH toks AS (SELECT ['1'] AS x, '1' as y UNION ALL SELECT ['2', '3'], null) SELECT ARRAY_CONCAT_AGG(x ORDER BY y) FROM toks`,
+			expectedRows: [][]interface{}{{
+				[]interface{}{"3", "2", "1"},
```
Comment (goccy) [inline query_test.go:682]: I tried this query in the BigQuery Console and got results `2, 3, 1` .
What do you think about this?

Comment (ohaibbq) [reply query_test.go:682]: `2, 3, 1` seems correct. I'll take a look at what's going on.

**Comment by ohaibbq** [comment]
It looks like there was also previously a bug with `ARRAY_CONCAT_AGG(... LIMIT 1)`. The `LIMIT` clause affects the number of arrays aggregated, not the number of elements in the result array.

**Review by goccy — APPROVED** [review]
Thank you for your awesome PR !! LGTM 👍

**Pull request merged**